align_system/algorithms/oracle_adm.py

```diff
@@ -0,0 +1,321 @@
+                                return 'Tourniquet'
+                        else:
+                            if 'Hemostatic gauze' in poss_treatments:
+                                return 'Hemostatic gauze'
+                    case 'Shrapnel':
+                        if 'face' in injury_location:
+                            if 'Nasopharyngeal airway' in poss_treatments:
+                                return 'Nasopharyngeal airway'
+                        else:
+                            if 'Hemostatic gauze' in poss_treatments:
+                                return 'Hemostatic gauze'
+                    case 'Internal':
+                        return 'Pain Medications'
+                    case _:
+                        raise ValueError(f"Unknown injury type: {injury_name}")
+
+                return None
+
+            # Determine what treatment to apply, narrow down applicable characters
+            if 'treatment' not in action_to_take.parameters:
+                # Can only apply treatments if supplies are available
+                poss_treatments = [s.type for s in scenario_state.supplies if s.quantity > 0]
+                ta3_server_treatments = [
+                    'Burn Dressing', 'Decompression Needle', 'Hemostatic gauze', 'Nasopharyngeal airway',
+                    'Pressure bandage', 'Splint', 'Tourniquet', 'Pain Medications'
+                ]
+                poss_treatments = [t for t in poss_treatments if t in ta3_server_treatments]
+
+                # Limit possible characters to those with injuries with specified location and supplies available
+                would_be_used = set()
+                if 'location' in action_to_take.parameters:
+                    tmp = list()
+                    for c in poss_chars:
+                        for injury in c.injuries:
+                            if injury.location == action_to_take.parameters["location"]:
+                                potential_treatment = _get_treatment(poss_treatments, injury.name, injury.location)
+                                if potential_treatment is not None:
+                                    tmp.append(c)
+                                    would_be_used.add(potential_treatment)
+                                    break
+                    poss_chars = tmp
+                else:  # Location not determined yet, could pick any treatment based on given supplies and injuries
+                    tmp = list()
+                    for c in poss_chars:
+                        for injury in c.injuries:
+                            potential_treatment = _get_treatment(poss_treatments, injury.name, injury.location)
+                            if potential_treatment is not None:
+                                tmp.append(c)
+                                would_be_used.add(potential_treatment)
+                                break
+                    poss_chars = tmp
+
+                if not would_be_used:
```

Comment:
I've seen the convention a few times here of `if not some_enumerable_thing` to check if it's empty or not, but I think it's much more clear to say `if len(some_enumerable_thing) == 0`.  Otherwise at a glance I'm not sure if `would_be_used` is a bool or what (and have to check elsewhere in the code to confirm).